File: EasyDarkMode/UIColor+DM.m
//
//  UIColor+DM.m
//  EasyInterface
//
//  Created by Elenion on 2019/8/5.
//  Copyright © 2019 Elenion. All rights reserved.
//

#import "UIColor+DM.h"
#import "DMManager.h"

@implementation UIColor (DM)

+ (UIColor *)dm_colorWithColorLight:(UIColor *)light dark:(UIColor *)dark {
#if __IPHONE_13_0
    if (@available(iOS 13.0, *)) {
        return [self colorWithDynamicProvider:^UIColor * _Nonnull(UITraitCollection * _Nonnull traitCollection) {
            switch (traitCollection.userInterfaceStyle) {
                case UIUserInterfaceStyleDark:
                    return dark;
                case UIUserInterfaceStyleLight:
                case UIUserInterfaceStyleUnspecified:
                default:
                    return light;
            }
        }];
    } else {
#endif
        switch (DMManager.shared.interfaceStyleForLowerSystem) {
            case DMUserInterfaceStyleDark:
                return dark;
            case DMUserInterfaceStyleLight:
            case DMUserInterfaceStyleUnspecified:
            default:
                return light;
        }
#if __IPHONE_13_0
    }
#endif
}

@end

File: EasyDarkMode+Swift/UIColor+DM.swift
//
//  UIColor+DM.swift
//  EasyDarkMode
//
//  Created by Elenion on 2019/8/22.
//  Copyright © 2019 Elenion. All rights reserved.
//

import UIKit

extension UIColor: InterfaceCompatable {
    
}

public extension InterfaceAdaptor where Base == UIColor.Type {
    func color(light: UIColor, dark: UIColor) -> UIColor {
        return UIColor.dm_color(withColorLight: light, dark: dark)
    }
}

public extension InterfaceAdaptor where Base == UIColor.Type {
    @available(*, deprecated, renamed: "color(light:dark:)")
    func color(color light: UIColor, dark: UIColor?) -> UIColor? {
        return color(light: light, dark: dark ?? light)
    }
}
